ddtrace/contrib/aiohttp/middlewares.py

```diff
@@ -0,0 +1,96 @@
+from ..asyncio import context_provider
+from ...ext import AppTypes, http
+from ...compat import stringify
+
+
+class TraceMiddleware(object):
+    """
+    aiohttp Middleware class that will append a middleware coroutine to trace
+    incoming traffic.
+    """
+    def __init__(self, app, tracer, service='aiohttp-web'):
+        # safe-guard: don't add the middleware twice
+        if getattr(app, '__datadog_middleware', False):
+            return
+        setattr(app, '__datadog_middleware', True)
+
+        # keep the references
+        self.app = app
+        self._tracer = tracer
+        self._service = service
+
+        # the tracer must work with asynchronous Context propagation
+        self._tracer.configure(context_provider=context_provider)
+
+        # configure the current service
+        self._tracer.set_service_info(
+            service=service,
+            app='aiohttp',
+            app_type=AppTypes.web,
+        )
+
+        # add the async tracer middleware
+        self.app.middlewares.append(self.middleware_factory())
+        self.app.on_response_prepare.append(self.signal_factory())
+
+    def middleware_factory(self):
+        """
+        The middleware factory returns an aiohttp middleware that traces the handler execution.
+        Because handlers are run in different tasks for each request, we attach the Context
+        instance both to the Task and to the Request objects. In this way:
+            * the Task may be used by the internal tracing
+            * the Request remains the main Context carrier if it should be passed as argument
+              to the tracer.trace() method
+        """
+        async def middleware(app, handler):
+            async def attach_context(request):
+                # trace the handler
+                request_span = self._tracer.trace(
+                    'aiohttp.request',
+                    service=self._service,
+                    span_type=http.TYPE,
+                )
+
+                # attach the context and the root span to the request
+                request['__datadog_context'] = request_span.context
```

Comment:
In many cases the ``Context`` has been useful for users. Make it a generic available keyword like: ``datadog_context``